lib/simp/cli/commands/bootstrap.rb

```diff
@@ -1,66 +1,335 @@
+    "puppetlabs.services.puppet-admin.puppet-admin-service/puppet-admin-service": "/puppet-admin-api"
+    "puppetlabs.trapperkeeper.services.status.status-service/status-service": "/status"
+}
+EOM
+})
+        say "> Sucessfully wrote web-routes.conf to #{puppetserver_dir}/web-routes.conf".green
+      rescue => error
+        fail( "Failed to configure the puppetserver, with error #{error.message}" )
+      end
+
+      # Firstrun is tagged and run against the bootstrap puppetserver port, 8150.
+      # This run will configure puppetserver and puppetdb; all subsequent runs
+      # will run against the conifgured masterport.
+      say "> Running puppet agent, with --tags pupmod,simp".cyan
+      pupcmd = 'puppet agent --onetime --no-daemonize --no-show_diff --verbose --no-splay --masterport=8150 --ca_port=8150'
+      # Firstrun is tagged and run against the bootstrap puppetserver port, 8150.
+      linecounts << track_output("#{pupcmd} --tags pupmod,simp 2> /dev/null", '8150')
+
+      # If selinux is enabled, relabel the filesystem.
+      FileUtils.touch('/.autorelabel')
+      if Facter.value(:selinux) && !Facter.value(:selinux_current_mode).nil? && (Facter.value(:selinux_current_mode) != "disabled")
+        say "> Relabeling filesystem for selinux".cyan
+        @logfile.puts("Relabeling filesystem for selinux.\n")
+        system("fixfiles -f relabel 2>&1 | tee -a #{@logfile.path}")
+      end
+
+      # SIMP is not single-run idempotent.  Until it is, run puppet twice.
+      say "> Running puppet without tags".cyan
+      pupcmd = "puppet agent --onetime --no-daemonize --no-show_diff --verbose --no-splay"
+      # This is fugly, but until we devise an intelligent way to determine when your system
+      # is 'bootstrapped', we're going to run puppet in a loop.
+      (0..1).each do
+        track_output("#{pupcmd}")
+      end
+
+      # Clean up the leftover puppetserver process (if any)
+      begin
+        pserver_proc = %x{netstat -tlpn}.split("\n").select{|x| x =~ /\d:8150/}
+        unless pserver_proc.empty?
+          pserver_port = %x{puppet config print masterport}
+          # By this point, bootstrap has applied config settings to puppetserver.
+          # Don't kill puppetserver if it's configured it to listen on 8150.
+          unless (pserver_port == '8150')
+            pserver_pid = pserver_proc.first.split.last.split('/').first.to_i
+            Process.kill('KILL',pserver_pid)
+          end
+        end
+      rescue Exception => e
+        say e
+        say "> The bootstrap puppetserver process running on port 8150 could not be killed. Please check your configuration!".magenta
+      end
+
+      # Print closing banner
+      say "> SIMP Bootstrap Complete!".yellow
+      say "> Duration of complete bootstrap: #{Time.now - bootstrap_start_time} seconds"
+      if !system('ps -C httpd > /dev/null 2>&1') && (linecounts.include?(-1) || (linecounts.uniq.length < linecounts.length))
```

Comment:
(linecounts.uniq.length < linecounts.length) <-- what is this checking?  Not at all obvious.